ee/src/main/java/org/jboss/as/ee/concurrent/ControlPointUtils.java

```diff
@@ -217,7 +228,7 @@ public T call() throws Exception {
                         }
                     }
                 } catch (Exception e) {
-                    EeLogger.ROOT_LOGGER.failedToRunTask(e);
+                    EeLogger.ROOT_LOGGER.failedToRunTask(callable,e);
```

Comment:
Exceptions thrown by Callable.call() also need to be propagated to the caller.  Thus the same treatment as above is needed here.

Reply:
It is, two lines bellow: 
 } catch (Exception e) {
                    EeLogger.ROOT_LOGGER.failedToRunTask(callable,e);
                }
                throw EeLogger.ROOT_LOGGER.cannotRunScheduledTask(callable);

Reply:
There are two problems with that handling:

1. It generates an incorrect exception message.
2. It obfuscates the exception thrown by the caller.

e.g. If the control point allows execution, but the subsequent invocation of callable.call() throws an exception, the ControlledScheduledCallable.call() method will throw a wrapped IllegalStateException to the caller containing the message "Cannot run scheduled task %s as container is suspended".  The message is not correct, since the container was *not* suspended (1).  Additionally, the checked exception thrown by callable.call() is incorrectly wrapped as a IllegalStateException (2).

Reply:
#1 - how can it now generate incorrect exception? This code was there before, only change is another param, to point what kind of task failed.
#2 - again, this has been there way before, has nothing to do with this PR.

Reply:
1. I explained this in my previous message.  See the e.g. for details.
2. This PR aims to fix error propagation for scheduled tasks, does it not?  Thus we should address both run() and call() scenarios.

Reply:
Sort of, but only for ControlledScheduledRunnable. I got tad thrown away by: Additionally, the checked exception thrown by callable.call() is incorrectly wrapped as a IllegalStateException (2) - since it isnt. Original exception is being logged and new one is thrown, though with wrong message.

Something like: 
try {
                    if (controlPoint.beginRequest() == RunResult.RUN) {
                        try {
                            return callable.call();
                        } finally {
                            controlPoint.requestComplete();
                        }
                    }
                } catch (Exception e) {
                    EeLogger.ROOT_LOGGER.failedToProcureCallableResult(callable,e);
                    throw e;
                }
                throw EeLogger.ROOT_LOGGER.cannotRunScheduledTask(callable);
                
should be enough?

Reply:
> Original exception is being logged and new one is thrown

This is what I mean by "incorrectly wrapped as an ISE".  Callable.call() should throw the unadulterated checked exception - it should not wrap it as a runtime exception.

Any exception thrown by ControlPoint *should* be wrapped as a runtime exception, but any exception thrown by Callable.call() should be rethrown to the caller without wrapping.  For the latter, we do not want a redundant log message.